File: frontend-web/frontend_website/src/types/employee.ts
export interface Employee {
  employeeId: string;
  firstName: string;
  lastName: string;
  email: string;
  gender?: string;
  hireDate: string;
  dateOfBirth?: string;
  address?: string;
  phoneNumber?: string;
  maritalStatus?: string;
  status: string;
  departmentId?: string;
  departmentName?: string;
  jobId?: string;
  jobName?: string;
  roleId: string;
  roleName: string;
}

export interface Department {
  departmentId: string;
  departmentName: string;
}

export interface JobTitle {
  jobId: string;
  jobName: string;
  jobDescription?: string;
  payGrade?: string;
}

export interface Role {
  roleId: string;
  roleName: string;
} 

File: backend/workforce/src/main/java/cit/edu/workforce/Entity/Employee.java
package cit.edu.workforce.Entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDate;
import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "employee")
public class Employee {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(name = "employee_id", updatable = false, nullable = false)
    private UUID employeeId;

    @Column(name = "first_name", nullable = false)
    private String firstName;

    @Column(name = "last_name", nullable = false)
    private String lastName;

    @Column(nullable = false, unique = true)
    private String email;

    @Column
    private String gender;

    @Column(name = "hire_date")
    private LocalDate hireDate;

    @Column(name = "birth_date")
    private LocalDate birthDate;

    @Column
    private String address;

    @Column
    private String phone;

    @Column
    private String status = "ACTIVE";  // Default status

    @OneToOne
    @JoinColumn(name = "user_id")
    private UserAccount userAccount;

    @ManyToOne
    @JoinColumn(name = "role_id")
    private Role role;

    @ManyToOne
    @JoinColumn(name = "job_id")
    private JobTitle jobTitle;

    @ManyToOne
    @JoinColumn(name = "department_id")
    private Department department;
} 